quinn-h3/src/qpack/prefix_string/encode.rs

```diff
@@ -93,6 +75,32 @@ impl HuffmanEncoder {
     }
 }
 
+fn set_bits(out: &mut [u8], pos: &BitWindow, value: u8) {
+    debug_assert!(pos.bit < 8);
+    debug_assert!(pos.count <= 8);
+    debug_assert!(pos.count > 0);
+
+    if (pos.bit + pos.count) <= 8 {
+        debug_assert_eq!(out[pos.byte as usize] | PAD_LEFT[pos.bit as usize], 255);
+        let pad_left = out[pos.byte as usize] | PAD_RIGHT[(8 - pos.bit) as usize];
+        let shifted = value << (8 - pos.bit - pos.count) | PAD_LEFT[pos.bit as usize];
+        let pad_right = PAD_RIGHT[(8 - pos.count - pos.bit) as usize];
+        out[pos.byte as usize] = (pad_left & shifted) | pad_right;
```

Comment:
Is it intended that if bit = 0 and count = 8, `value` is completely ignored? I'm probably missing something here.

Reply:
Pretty sure it isn't? I think the first part of the test covers this case.

Reply:
As in this case:

```
pos.count = 8;
set_bits(&mut buf, &pos, 0b1_0101);
assert_eq!(&buf[..1], &[0b1_0101]);
pos.byte += 1;
```

Reply:
Er, sorry, I meant when bit = 0 and count = 0.

Reply:
For example:

```rust
let mut buf = [0; 1];
set_bits(
    &mut buf,
    &BitWindow {
        byte: 0,
        bit: 0,
        count: 1,
    },
    0,
);
```

produces `buf == [255]`, which doesn't seem right, unless I'm wildly misunderstanding the intended semantics (which should probably be commented?)